addons/base/views.py

```diff
@@ -315,7 +315,7 @@ def get_auth(auth, **kwargs):
                 if not node.is_contributor(auth.user):
                     download_is_from_mfr = request.headers.get('X-Cos-Mfr-Render-Request', None)
                     # version index is 0 based
-                    version_index = version - 1
+                    version_index = int(version) - 1
```

Comment:
This probably fixes it, but I'd be concerned that other code in the future will rely on the `version` value being a int. Right now, it starts as `Union[str, None]`, then gets reassigned ~L304 as `Union[str, int , None]`, which is confusing.

Why not ensure that `version` is either `str` or `None` when it first gets assigned (~L292)?

Reply:
`version` can be a non-castable string for non-osfstorage providers (e.g. `latest` for DV).  If you do that, you might want to move initialization inside the `if provider_name == 'osfstorage':` block.

Reply:
Moved the `int` cast under the `osfstorage` check when it's first accessed (version isn't actually used anywhere outside the `osfstorage` block it seems anyhow!

Reply:
@sloria the reason I didn't do it there was because @fitz mentioned that version can be a non-castable string for non-osfstorage providers, and line 292 is before we are checking for `osfstorage`